nessie/src/main/java/org/apache/iceberg/nessie/TableReference.java

```diff
@@ -21,25 +21,28 @@
 
 import java.time.Instant;
 import java.util.List;
+import javax.annotation.Nonnull;
 import org.apache.iceberg.catalog.TableIdentifier;
 import org.apache.iceberg.relocated.com.google.common.base.Splitter;
 
 public class TableReference {
 
   private static final Splitter BRANCH_NAME_SPLITTER = Splitter.on("@");
+  @Nonnull
```

Comment:
what does this do?

Reply:
Been requested earlier during review to avoid unnecessary null-checkes